File: src/main/java/fi/quanfoxes/lexer/ComparisonOperator.java
package fi.quanfoxes.lexer;

public class ComparisonOperator extends Operator {
    private ComparisonOperator counterpart;

    public ComparisonOperator(String identifier, int priority) {
        super(identifier, OperatorType.COMPARISON, priority);
    }

    public ComparisonOperator setCounterpart(ComparisonOperator counterpart) {
        this.counterpart = counterpart;
        return this;
    }

    public ComparisonOperator getCounterpart() {
        return counterpart;
    }
}

File: Vivid/Lexer/ComparisonOperator.cs
using System;

public class ComparisonOperator : Operator
{
	public ComparisonOperator? Counterpart { get; set; }

	public ComparisonOperator(string identifier, int priority) : base(identifier, OperatorType.COMPARISON, priority) { }

	public ComparisonOperator SetCounterpart(ComparisonOperator counterpart)
	{
		Counterpart = counterpart;
		return this;
	}

	public override bool Equals(object? other)
	{
		return ReferenceEquals(this, other);
	}

	public override int GetHashCode()
	{
		return HashCode.Combine(base.GetHashCode(), Counterpart?.Identifier);
	}
}